### PR TITLE
Proposal of new docs after WAF fixes

### PR DESCRIPTION
**Pull request by radixo**
Product Version(s):

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 1]
```diff
@@ -178,39 +178,36 @@ Now if a SQL Injection attack is detected for rule ID 942100, you will see the g
...
 
-#### Deny traffic
+#### Allow traffic
 
-For example, if you want to deny traffic with a 403 response, do the following steps: 
+For example, if you want to allow traffic, do the following steps:
 
-1. Modify the modsecurity core rulesets' `SecDefaultAction` to deny rules by commenting and uncommenting these sections in the `crs-setup.conf`, 
-which will change the `SecDefaultAction` from `pass` to `deny`.
+1. Change the disruptive action, on file `RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf.example` section `'Changing the Disruptive Action for Anomaly Mode'`.
```
Comment (girishshankaran) [inline calico-enterprise/threat/web-application-firewall.mdx]: This step is unclear.

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 2]
```diff
@@ -274,7 +274,9 @@ curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modse
 curl https://raw.githubusercontent.com/coreruleset/coreruleset/v3.3/dev/crs-setup.conf.example > crs-setup.conf
 ```
 
-3. Change your current directory to the my-ruleset folder where your core rules set files live. Create a configMap containing all the files downloaded into your new directory and replace the existing rule set with it:
+3. Do your customized settings on the default files, probably on the `.example` files and after renaming, like previously on this doc.
```
Comment (girishshankaran) [inline calico-enterprise/threat/web-application-firewall.mdx]: Do your customized settings on the default files, probably on the `.example` files after renaming (refer to the earlier step).

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 3]
```diff
@@ -274,7 +274,9 @@ curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modse
...
+
+4. Still at the my-ruleset folder where your core rules set files live. Create a configMap containing all the files downloaded into your new directory and replace the existing rule set with it:
```
Comment (girishshankaran) [inline calico-enterprise/threat/web-application-firewall.mdx]: Go to `my-ruleset` folder where your core rules set files are located, create a ConfigMap containing all the files downloaded into your new directory and replace the existing rule set with it.

**Review by girishshankaran — CHANGES_REQUESTED** [review]
Since these changes are applicable for both 3.17 and 3.18, please include them in 3.17 folder as well.

File: calico-enterprise/threat/web-application-firewall.mdx (hunk 3 above)
Comment (girishshankaran) [reply calico-enterprise/threat/web-application-firewall.mdx]: Or, we can say? Go to the `my-ruleset` folder

File: calico-enterprise/threat/web-application-firewall.mdx (hunk 1 above)
Comment (electricjesus) [reply calico-enterprise/threat/web-application-firewall.mdx]: I think this should be 'Anomaly scores could result in Disruptive action. To change the disruption action, see the file `RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf.example` section `'Changing the Disruptive Action for Anomaly Mode'`'

File: calico-enterprise/threat/web-application-firewall.mdx (hunk 2 above)
Comment (electricjesus) [reply calico-enterprise/threat/web-application-firewall.mdx]: "If you wish to customize settings on default included rules, please see .example files for guidance. You must save these files without the .example extension if you wish for these changes to take effect."

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 4]
```diff
@@ -274,7 +274,9 @@ curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modse
 curl https://raw.githubusercontent.com/coreruleset/coreruleset/v3.3/dev/crs-setup.conf.example > crs-setup.conf
 ```
 
-3. Change your current directory to the my-ruleset folder where your core rules set files live. Create a configMap containing all the files downloaded into your new directory and replace the existing rule set with it:
+3. If you wish to customize settings on default included rules, please see `.example` files for guidance. You must save these files without the .example extension if you wish for these changes to take effect.
```
Comment (girishshankaran) [inline calico-enterprise/threat/web-application-firewall.mdx]: If you wish to customize settings on default included rules, see `.example` files for guidance.

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 5]
```diff
@@ -251,12 +248,15 @@ ModSecurity provides the following actions that you update in SecAction:
...
 mkdir my-ruleset && cd my-ruleset
-curl -O https://raw.githubusercontent.com/coreruleset/coreruleset/v3.3/dev/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf
+curl -L https://github.com/coreruleset/coreruleset/archive/refs/heads/v3.4/dev.zip -o crs.zip
+unzip -j crs.zip 'coreruleset-3.4-dev/rules/*'
+rm crs.zip
 ```
+
 2. Download bootstrap files.
 
   Download the following required bootstrapping configuration files [Reference](https://medium.com/lightbaseio/web-application-firewall-in-go-feat-owasp-modsecurity-core-rule-set-3f97a26e3311) that are used by Tigera operator:
```
Comment (girishshankaran) [inline calico-enterprise/threat/web-application-firewall.mdx]: Download the following required bootstrapping configuration files from this [site]( https://medium.com/lightbaseio/web-application-firewall-in-go-feat-owasp-modsecurity-core-rule-set-3f97a26e3311). These files are used by Tigera operator.

**Review by girishshankaran — COMMENTED** [review]
Please make these changes to the 3.17 folder as well. You can just copy the file.

**Review by girishshankaran — APPROVED** [review]
LGTM

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 6]
```diff
@@ -178,39 +178,36 @@ Now if a SQL Injection attack is detected for rule ID 942100, you will see the g
 
 #### Change WAF rules
 
-By default, the WAF OWASP Core Rule set is configured to log a warning if there is an attack. However, if you want traffic for the service to also be denied or dropped, 
+By default, the WAF OWASP Core Rule set is configured to log all matched rules and block offending requestis using Anomaly Scoring Mode. However, if you want traffic for the service to not be blocked and only logged,
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:181]:
```suggestion
By default, the WAF OWASP Core Rule set is configured to log all matched rules and block offending request is using Anomaly Scoring Mode. However, if you want traffic for the service to not be blocked and only logged,
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 7]
```diff
@@ -178,39 +178,36 @@ Now if a SQL Injection attack is detected for rule ID 942100, you will see the g
...
+3. Run these commands:
 
 ```bash
   kubectl create cm --dry-run=client --from-file=. -o yaml -n tigera-operator modsecurity-ruleset > ../my-ruleset.yaml
   kubectl replace -f ../my-ruleset.yaml
 ```
 
-Note that switching to deny means the request is not processed and a 403 forbidden error is returned when a rule is triggered. 
-All denied traffic means that any HTTP request will return HTTP 403 Response Code from Envoy to the originating service. 
+Note that switching to pass means the request is now processed and the 403 forbidden error is gone even if a rule is triggered.
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:210]:
```suggestion
Note that switching to pass means the request is now processed and the 403 forbidden error is gone, even if a rule is triggered.
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 8]
```diff
@@ -274,7 +274,9 @@ curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modse
 curl https://raw.githubusercontent.com/coreruleset/coreruleset/v3.3/dev/crs-setup.conf.example > crs-setup.conf
 ```
 
-3. Change your current directory to the my-ruleset folder where your core rules set files live. Create a configMap containing all the files downloaded into your new directory and replace the existing rule set with it:
+3. If you wish to customize settings on default included rules, see `.example` files for guidance. You must save these files without the .example extension if you wish for these changes to take effect.
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:277]:
```suggestion
3. To customize default core rules, see the `.example` files for guidance.  For changes to take effect, ensure that you save the file without the .example extension.
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 9]
```diff
@@ -274,7 +274,9 @@ curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modse
...
+
+4. Go to my-ruleset folder where your core rules set files are located, create a ConfigMap containing all the files downloaded into your new directory and replace the existing rule set with it.
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:279]:
```suggestion
4. Go to the `my-ruleset` folder where your core rules set files are located, create a ConfigMap containing all the files downloaded into your new directory, and replace the existing rule set with it.
```

**Review by bmckercher123 — CHANGES_REQUESTED** [review]
Minor changes. Are these changes only for CE 3.18? What about CC?

**Review by electricjesus — APPROVED** [review]

File: calico-enterprise/threat/web-application-firewall.mdx (hunk 6 above)
Comment (bmckercher123) [reply calico-enterprise/threat/web-application-firewall.mdx:181]: Fixed in /main.

**Comment by bmckercher123** [comment]
All of the items have been addressed in WAF updates for 3.18. No need to merge. Closing out.

**Pull request closed without merging**